File: 908 - Smallest Range I.cpp
class Solution {
public:
    int smallestRangeI(vector<int>& A, int K) {
        int min = A[0], max = A[0];
        for (int a : A) {
            min = min < a ? min : a;
            max = max > a ? max : a;
        }
        int res = max - min;
        if (res >= 2 * K) {
            res -= 2 * K;
        } else {
            res = 0;
        }
        return res;
    }
};

File: 908 - Smallest Range I.py
class Solution:
    def smallestRangeI(self, A: list[int], K: int) -> int:
        minval = min(A)
        maxval = max(A)
        res = maxval - minval
        if res >= K * 2:
            res -= K * 2
        else:
            res = 0
        return res